src/electronic-ids/pkcs11/Pkcs11ElectronicID.cpp

```diff
@@ -26,9 +26,43 @@
 
 #include <map>
 
+#ifdef _WIN32
+#undef UNICODE
+#include <Shlobj.h>
+#include <Shlwapi.h>
+#include <Knownfolders.h>
+#elif defined(__APPLE__)
+#include <unistd.h>
+#endif
+
 namespace
 {
 
+std::string lithuanianPKCS11Path() {
+#ifdef _WIN32
+    PWSTR programFilesX86 = 0;
+    SHGetKnownFolderPath(FOLDERID_ProgramFilesX86, 0, nullptr, &programFilesX86);
+    std::wstring path = programFilesX86;
+    CoTaskMemFree(programFilesX86);
+    if (PathFileExistsW((path + L"\\PWPW\\pwpw-card-pkcs11.dll").c_str()))
+        path += L"\\PWPW\\pwpw-card-pkcs11.dll";
+    else
+        path += L"\\CryptoTech\\CryptoCard\\CCPkiP11.dll";
+    int len = WideCharToMultiByte(CP_UTF8, 0, path.data(), int(path.size()), nullptr, 0, nullptr, nullptr);
+    std::string out(size_t(len), 0);
+    WideCharToMultiByte(CP_UTF8, 0, path.data(), int(path.size()), &out[0], len, nullptr, nullptr);
+    return out;
+#elif defined(__APPLE__)
+    static const std::string path1("/Library/Security/tokend/CCSuite.tokend/Contents/Frameworks/libccpkip11.dylib");
+    static const std::string path2("/Library/PWPW-Card/pwpw-card-pkcs11.so");
+    return access(path1.c_str(), F_OK) == 0 ? path1 : path2;
+#elif defined(__LP64__)
```

Comment:
As far as I understand this is not needed. In a 64-bit Ubuntu, there is no `/usr/lib64`, as `/usr/lib` is already 64-bit by default. Here's how it looks in my machine:

```sh
$ tree -L 1 /usr/
/usr/
├── bin
├── games
├── include
├── lib
├── lib32
├── libexec
├── local
├── sbin
├── share
└── src
```

This was necessary when the OS and `/usr/lib` was 32-bit, but the installed driver was 64-bit. In this case  `/usr/lib64` exists and is for the non-default 64-bit libraries. But we don't support 32-bit OSes any more.

Please prove me wrong.

Reply:
We did investigate this with Heikki, and for some reason `lib64` did exist in his 64-bit Ubuntu 20.04 and the driver was installed there. So we have two cases in 64-bit systems: either the driver is installed to `lib` or `lib64`.

Reply:
We simply need to check if `/usr/lib64/pwpw...` is present and use it from there and from `/usr/lib/pwpw...` if not.